### PR TITLE
MDTZ-1749: Implement the simplified `jiraDesignInfoProvider` module

### PR DESCRIPTION
**Pull request by akostevich-atlassian**
### Context

The existing `jiraDesignInfoProvider` module specification includes the following primary *Actions*:
- `associateEntity`
- `disassociateEntity`
- `onEntityAssociated` (currently, unused by the app due to certain limitations)
- `onEntityDisassociated` (currently, unused by the app due to certain limitations)

Currently, by design and due to the specifics of the app, "_Figma for Jira_" implements the following  *Actions* and corresponding responsibilities:

| Action                   | Responsibilities                                                                                                                                                                                             |
|:-------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `associateEntity`        | - Fetch a *Design* from Figma.<br> - Create an *Association* with a Jira *Issue*.<br> - Submit a *Design* with to Jira.<br> - Handle backward integration with Figma (update Jira *Issue Properties*, create a Figma *Dev Resource*). |
| `disassociateEntity`     | - Delete an *Association* with a Jira *Issue*.<br> - Submit a *Design* to Jira.<br> - Handle backward integration with Figma (update Jira *Issue Properties*, delete a Figma *Dev Resource*). |

The simplified `jiraDesignInfoProvider` module delivers new and improved _Actions_, which allows to move some responsibilities from the app to Jira and handle the remaining ones more granularly.

| Action               | Responsibilities                                                                                         |
|:------------------------|:---------------------------------------------------------------------------------------------------------|
| `getEntityByUrl`              | - Fetch and return a *Design* from Figma<br>                                                             |
| `onEntityAssociated`      | - Handle backward integration with Figma (update Jira *Issue Properties*, create a Figma *Dev Resource*) |
| `onEntityDisassociated` | - Handle backward integration with Figma (update Jira *Issue Properties*, delete a Figma *Dev Resource*) |

As the result, the simplified `jiraDesignInfoProvider` module will bring the following benefits to the app:
- Less and simpler code to maintain
- Less points of failure and better reliability

### Changes

- **feat:** Implements the simplified `jiraDesignInfoProvider` module specification. The PR adds the new _Actions_ (`getEntityByUrl`, `onEntityAssociated` and `onEntityDisassociated`) without affecting the existing ones (`associateEntity` and  `disassociateEntity`) to enable the gradual rollout of the changes. Therefore, the changes will have **no effect** until the feature flag is enabled on the Atlassian platform side.

### Notes
- The legacy `associateEntity` and  `disassociateEntity` _Actions_ will be removed from the app when the rollout of the simplified `jiraDesignInfoProvider` module is completed.

### CODE REVIEW

**Comment by francox9** [comment]
🔥

File: README.md [hunk 1]
```diff
@@ -144,26 +144,39 @@ If needed, you could test these APIs directly by mimicking Jira backend.
    ```
 
    ```shell
-   curl --request POST \
-     --url '${APP_URL}/entities/associateEntity?userId=${ATLASSIAN_USER_ID}' \
+   curl --request PUT \
```
Comment (stannnous) [inline README.md:147]: Did we agree on POST for this action?

File: src/web/routes/entities-v2/entities-router.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+entitiesRouterV2.use(jiraServerToServerSymmetricJwtAuthenticationMiddleware);
+
+/**
+ * Returns Figma design details for the given URL.
+ * Invoked when an entity is being associated with an Atlassian resource (e.g. a Jira Issue).
+ *
+ * @see https://developer.atlassian.com/cloud/devops-provider-actions/specifications/current/#get-entity-by-url
+ */
+entitiesRouterV2.put(
```
Comment (stannnous) [inline src/web/routes/entities-v2/entities-router.ts:40]: post

File: src/web/routes/entities-v2/schemas.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+					properties: {
+						ari: { type: 'string' },
+						id: { type: 'string' },
+					},
+					required: ['ari', 'id'],
+				},
+				associateWith: {
+					type: 'object',
+					properties: {
+						ati: { type: 'string', pattern: 'ati:cloud:jira:issue' }, // Handle only associations with a Jira Issue.
```
Comment (stannnous) [inline src/web/routes/entities-v2/schemas.ts:59]: Nice @akostevich-atlassian+++++

File: src/web/routes/entities-v2/entities-router.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+		// Jira Frontend appends this query parameter to the design URL as a temporary workaround that allows to
+		// distinguish "Backfill" from normal "Associate" operations. Therefore, the app can apply some special considerations
+		// to designs for backfill (e.g., handle deleted designs).
+		const isBackfill = designUrl.searchParams.has(
+			'com.atlassian.designs.backfill',
+			'true',
+		);
+
+		if (isBackfill) {
+			getDesignByUrlForBackfillUseCase
```
Comment (stannnous) [inline src/web/routes/entities-v2/entities-router.ts:69]: To clarify: the Figma app will return the minimal design for the backfill use case which will be ingested with the issue association by Jira and the app will then async send the full design (without the issue association). This is temporary until the backfill use case can be removed after the Figma widget is updated.

**Review by stannnous — CHANGES_REQUESTED** [review]
Great work. Just the HTTP method change for getEntityByUrl

File: README.md (hunk 1 above)
Comment (akostevich-atlassian) [reply README.md:147]: We did -- I need to fix this.

**Review by stannnous — APPROVED** [review]

**Review by knguyen-figma — APPROVED** [review]

**Pull request merged**